File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    people.sort()
    ans = 0
    start = 0
    end = len(people)-1
    while start<=end:
        if people[start]+people[end]<=limit:
            ans +=1
            start += 1
            end -= 1
        else:
            end -= 1
            ans += 1
    # if len(people)%2 == 1:
    #     ans += 1
        
    return ans

File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
    public int solution(int[] people, int limit) {
        int boat = 0;
        int sum = 0;
        
        Arrays.sort(people);
        int l = 0;
        int r = people.length-1;
        while(l<r){
            int cur = people[l]+people[r];
            if(cur>limit){
                r -= 1;
                boat += 1;
            }
            else{
                boat += 1;
                l += 1;
                r -= 1;
            }
        }
        //홀수면 하나 증가
        if(l==r){
            return boat+1;
        }
        else{
            return boat;
        }
    }
}